crates/scarb-api/src/artifacts/deserialized.rs

```diff
@@ -0,0 +1,39 @@
+use anyhow::{Context, Result};
+use camino::{Utf8Path, Utf8PathBuf};
+use serde::Deserialize;
+use std::fs;
+
+#[derive(Deserialize, Debug, PartialEq, Clone)]
+pub(super) struct StarknetArtifacts {
+    pub version: u32,
+    pub contracts: Vec<StarknetContract>,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize, Debug, PartialEq, Clone)]
+pub(super) struct StarknetContract {
+    pub id: String,
+    pub package_name: String,
+    pub contract_name: String,
+    pub artifacts: StarknetContractArtifactPaths,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize, Debug, PartialEq, Clone)]
+pub(super) struct StarknetContractArtifactPaths {
+    pub sierra: Utf8PathBuf,
+}
+
+/// Get deserialized contents of `starknet_artifacts.json` file generated by Scarb
+///
+/// # Arguments
+///
+/// * `path` - A path to `starknet_artifacts.json` file.
+pub(super) fn artifacts_for_package(path: &Utf8Path) -> Result<StarknetArtifacts> {
```

Comment:
It is only used in `represensation.rs`, no? Why not put it there?

Reply:
This deserializes into structs defined above. I though it makes sense to have it along the structs it uses.